### DISCUSSION TITLE
Setup chat with internet search

### DISCUSSION THREAD

**Discussion by WitnessMee** [danny-avila/LibreChat#3192] [Q&A] [answered]
OpenAI API is unable to do internet searches. Is it possible to setup LibreChat to do the searches and send the results the openai api?

I have setup [Google Search](https://www.librechat.ai/docs/configuration/tools/google_search) but that doesn't seem to add search capabilities inside the chat. 
When I ask a question like "Can you search the internet for..." I get this response "I'm unable to browse the internet"

**Comment by WitnessMee** [comment, marked as answer]
I was able to do a web search in LibreChat, using a plugin:
Dropdown top left --> Plugins. A new dropdown will appear. Go to Plugin store --> Web Search

This is great. However, I am still wondering what the use case for the Google Search setup is.

**Reply by prononext** [reply]
@WitnessMee did you activate the plugin after installing it? If not it will not be available for the open conversation. 

Also for Google Search you need to set API Keys - https://www.librechat.ai/docs/configuration/tools/google_search
```
      - GOOGLE_SEARCH_API_KEY=
      - GOOGLE_CSE_ID=
```

At my end its working find just using the promt you mentioned. 
The only thing is that I would like to see more details of the results it got in the plugin request.

**Reply by WitnessMee** [reply]
I did set up the API keys correctly, as explained in that link. My second question was, why did I set it up?
Does the **Web Search** plugin use this setup? Or perhaps another setup?

In any case, using the Web Search plugin solves my problem.

**Reply by fuegovic** [reply]
@WitnessMee 
`GOOGLE_SEARCH_API_KEY` and `GOOGLE_CSE_ID` are used with this tool/plugin:
![image](https://github.com/danny-avila/LibreChat/assets/32828263/1ccefb0b-14fb-430b-82b2-e1a6c4a654a2)

**Reply by wupengfei0904** [reply]
The Google search plugin doesn't return enough content, it's all returned in summary form and it's hard to get some useful news unless a web page loader is added.
